src/main/java/io/pinecone/model/CreateIndexRequest.java

```diff
@@ -0,0 +1,197 @@
+package io.pinecone.model;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.pinecone.PineconeValidationException;
+
+public class CreateIndexRequest {
+    private String indexName;
+
+    private Integer dimension;
+
+    private String metric = "cosine";
+
+    private Integer pods = 1;
+
+    private Integer replicas = 1;
+
+    private String podType = "p1.x1";
+
+    private IndexMetadataConfig metadataConfig;
+
+    private String sourceCollection;
+
+    public CreateIndexRequest() {
+    }
+
+    public CreateIndexRequest indexName(String name) {
+        this.indexName = name;
+        return this;
+    }
+
+    /**
+     * The name of the index to be created. The maximum length is 45 characters.
+     * @return name
+     **/
+    @javax.annotation.Nullable
+    public String getIndexName() {
+        return indexName;
+    }
+
+    public void setIndexName(String indexName) {
+        this.indexName = indexName;
+    }
+
+    public CreateIndexRequest dimension(Integer dimension) {
+        this.dimension = dimension;
+        return this;
+    }
+
+    /**
+     * The dimensions of the vectors to be inserted in the index
+     * @return dimension
+     **/
+    @javax.annotation.Nonnull
+    public Integer getDimension() {
+        return dimension;
+    }
+
+    public void setDimension(Integer dimension) {
+        this.dimension = dimension;
+    }
+
+    public CreateIndexRequest metric(String metric) {
+        this.metric = metric;
+        return this;
+    }
+
+    /**
+     * The distance metric to be used for similarity search. You can use &#39;euclidean&#39;, &#39;cosine&#39;, or &#39;dotproduct&#39;.
+     * @return metric
+     **/
+    @javax.annotation.Nullable
```

Comment:
Same question on this as above for index name. Should this be Nullable?

Reply:
Since metric is already set to cosine by default, the nullable is expected. In the api docs, only index name and dimension are set to be required fields.